indico/modules/rb_new/client/js/components/modals/BlockingModal.jsx

```diff
@@ -0,0 +1,253 @@
+ *
+ * Indico is free software; you can redistribute it and/or
+ * modify it under the terms of the GNU General Public License as
+ * published by the Free Software Foundation; either version 3 of the
+ * License, or (at your option) any later version.
+ *
+ * Indico is distributed in the hope that it will be useful, but
+ * WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the GNU
+ * General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with Indico; if not, see <http://www.gnu.org/licenses/>.
+ */
+
+import React from 'react';
+import {connect} from 'react-redux';
+import PropTypes from 'prop-types';
+import {Button, Divider, Form, Grid, Message, Icon, Modal} from 'semantic-ui-react';
+import {Form as FinalForm, Field} from 'react-final-form';
+import setFieldTouched from 'final-form-set-field-touched';
+import createDecorator from 'final-form-calculate';
+import {Translate} from 'indico/react/i18n';
+import {ReduxFormField, formatters} from 'indico/react/forms';
+import PrincipalSearchField from 'indico/react/components/PrincipalSearchField';
+import DatePeriodField from 'indico/react/components/DatePeriodField';
+import RoomSelector from '../RoomSelector';
+import {createBlocking} from '../../actions';
+
+
+function validate({period, reason, rooms}) {
+    const errors = {};
+    if (period === undefined || !period.length) {
+        errors.period = Translate.string('Please choose a valid period!');
+    }
+    if (!reason) {
+        errors.reason = Translate.string('Please provide the reason for the blocking!');
+    }
+    if (rooms === undefined || !rooms.length) {
+        errors.rooms = Translate.string('Please choose at least one room for this blocking!');
+    }
+    return errors;
+}
+
+const formDecorator = createDecorator({
+    field: 'period',
+    updates: (periodValue) => {
+        return {
+            start_date: periodValue[0],
+            end_date: periodValue[1]
+        };
+    }
+}, {
+    field: 'rooms',
+    updates: (rooms) => {
+        return {
+            room_ids: rooms.map((room) => room.id)
+        };
+    }
+}, {
+    field: 'allowed',
+    updates: (allowed) => {
+        return {
+            allowed_principals: allowed.map((obj) => ({
+                id: obj.id,
+                is_group: obj.is_group
+            }))
+        };
+    }
+});
+
+
+class BlockingModal extends React.Component {
+    static propTypes = {
+        open: PropTypes.bool.isRequired,
+        onClose: PropTypes.func.isRequired,
+        rooms: PropTypes.array,
+        createBlocking: PropTypes.func.isRequired
+    };
+
+    static defaultProps = {
+        rooms: []
+    };
+
+    createBlocking = async (formData) => {
+        await this.props.createBlocking(formData); // eslint-disable-line react/destructuring-assignment
```

Comment:
check for the error handling logic in `submitBooking`. assuming you have matching field names here and in the webargs schema on the server, returning the error here will make final-form automatically show the server-side error on the corresponding field.